src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SingleSpaceSeparatorCheck.java

```diff
@@ -138,7 +140,7 @@ public void beginTree(DetailAST rootAST) {
     private void visitEachToken(DetailAST node) {
         DetailAST sibling = node;
 
-        while (sibling != null) {
+        do {
```

Comment:
the if statement above the recursive call means either changing this to a do-while or removing the if-siblingCount check.
I changed this to a do-while.